diff --git a/build/src/auth/oauth2client.d.ts b/build/src/auth/oauth2client.d.ts
index 7080d080a99d8b9d5e43b9d615d50dadfe0a18e2..d173f5ba36093af3c2a21bdd1fc48c10e4343572 100644
--- a/build/src/auth/oauth2client.d.ts
+++ b/build/src/auth/oauth2client.d.ts
@@ -445,10 +445,9 @@ export declare class OAuth2Client extends AuthClient {
     /**
      * Refreshes the access token.
      * @param refresh_token Existing refresh token.
-     * @private
      */
-    protected refreshToken(refreshToken?: string | null): Promise<GetTokenResponse>;
-    protected refreshTokenNoCache(refreshToken?: string | null): Promise<GetTokenResponse>;
+    refreshToken(refreshToken?: string | null): Promise<GetTokenResponse>;
+    refreshTokenNoCache(refreshToken?: string | null): Promise<GetTokenResponse>;
     /**
      * Retrieves the access token using refresh token
      *
